ui/src/features/WorkspaceSettings/components/MembersSettings.tsx

```diff
@@ -3,87 +3,109 @@ import { useState } from "react";
 
 import {
   Button,
-  Checkbox,
   DropdownMenu,
   DropdownMenuContent,
   DropdownMenuItem,
   DropdownMenuTrigger,
+  Input,
 } from "@flow/components";
+import { useUser, useWorkspace } from "@flow/lib/gql";
 import { useT } from "@flow/lib/i18n";
 import { useCurrentWorkspace } from "@flow/stores";
 import { Role, UserMember } from "@flow/types";
 
 type Filter = "all" | Role;
 
-const roles = ["admin", "reader", "writer"];
+const roles: Role[] = Object.values(Role);
 
 const MembersSettings: React.FC = () => {
   const t = useT();
-  const [currentWorkspace] = useCurrentWorkspace();
-
+  const [currentWorkspace, setCurrentWorkspace] = useCurrentWorkspace();
+  const { addMemberToWorkspace, removeMemberFromWorkspace, updateMemberOfWorkspace } =
+    useWorkspace();
+  const { searchUser } = useUser();
+  const [email, setEmail] = useState<string>("");
   const [currentFilter, setFilter] = useState<Filter>("all");
+  const [error, setError] = useState<string | undefined>();
 
   const filters: { id: Filter; title: string }[] = [
     { id: "all", title: t("All") },
-    { id: "admin", title: t("Admin") },
-    { id: "reader", title: t("Reader") },
-    { id: "writer", title: t("Writer") },
+    { id: Role.Owner, title: t("Owner") },
+    { id: Role.Reader, title: t("Reader") },
+    { id: Role.Reader, title: t("Maintainer") },
+    { id: Role.Writer, title: t("Writer") },
   ];
 
   const members =
-    (currentWorkspace?.members?.filter(
-      m => "userId" in m && (currentFilter !== "all" ? m.role === currentFilter : true),
-    ) as UserMember[]) ?? [];
+    (
+      currentWorkspace?.members?.filter(
+        m => "userId" in m && (currentFilter !== "all" ? m.role === currentFilter : true),
+      ) as UserMember[]
+    ).sort((a, b) => a.user?.name.localeCompare(b.user?.name ?? "") ?? 0) ?? [];
```

Comment:
Personally I feel like until (unless?) we implement an option for alphabetized sorting (A-Z and Z-A) this might be confusing when you add a member and they don't get added to the bottom. Especially if the list is longer than 4 or 5 members.

Reply:
Hmm...true. So I was facing an issue where I update the user role and it'll come on top of the list so I added this automatic sorting...but you are right, What do you suggest we should do?

Reply:
Hmm I wonder if this is something inconsistent happening on the backend.... was it always on top? Or on refresh the order was correct? 
It might be a caching issue since the members list is a nested object, perhaps tanstack query's caching is messing up somehow.
Let me know if refreshing fixes the order (p.s. how are you testing this locally?)

Reply:
Manually adding the user and setting the new workspace as current Workspace

Reply:
i'll update the comment with my findings

Reply:
I am using the order from the server and it works okay except when changing the roles